tests/test_parallel_processing.py is non-deterministic

Description:
tests/test_parallel_processing.py::test_process_parallel_callback fails intermittently:

>       assert collected == [10, 20, 30]
E       assert [10, 30, 20] == [10, 20, 30]
E         At index 1 diff: 30 != 20

The test gives a callback function to process_parallel and saves every result into a list called collected. Then it checks that the list is exactly [10, 20, 30]:

process_parallel(_identity, [10, 20, 30], callback=_on_result, progress_bar=False)
assert collected == [10, 20, 30]
The problem is that process_parallel runs the items in parallel (it uses pool.apply_async(..., callback=callback)). When you run things in parallel, they don't always finish in the same order you sent them in — whichever one finishes first calls the callback first. So collected can end up in any order, like [10, 30, 20].

This is not a bug in process_parallel itself — that function still returns the results in the correct order at the end. The problem is only in the test: it shouldn't expect the callback to be called in a specific order, because parallel tasks can finish at any time.

Suggested fix:

Instead of checking the exact order, the test should just check that all three values are there (no matter the order). For example by sorting the list before comparing:

assert sorted(collected) == [10, 20, 30]

Happened on Python 3.13, happens only sometimes.

--- a/tests/test_parallel_processing.py
+++ b/tests/test_parallel_processing.py
@@ -51,7 +51,7 @@ def test_process_parallel_callback():
         collected.append(result)
 
     process_parallel(_identity, [10, 20, 30], callback=_on_result, progress_bar=False)
-    assert collected == [10, 20, 30]
+    assert sorted(collected) == [10, 20, 30]
 
 
 def test_process_parallel_empty_items():

